ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -626,12 +639,22 @@ S3ObjectListFree(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, char **objectL
  *     Add objects retrieved from S3 bucket into the object list, and allocate the memory needed.
  */
 static int
-S3ObjectListAdd(
-  S3_STORAGE *s3, char ***objectList, const std::vector<std::string> &objects, const uint32_t count)
+S3ObjectListAdd(const S3_STORAGE *s3, char ***objectList, const std::vector<std::string> &objects,
+  const uint32_t count)
 {
-    char **entries = (char **)malloc(sizeof(char *) * count);
+    char **entries;
+    if ((entries = (char **)malloc(sizeof(char *) * count)) == NULL) {
+        s3->log->LogVerboseMessage(
+          -3, "S3ObjectListAdd: unable to allocate memory for object list.");
+        return (ENOMEM);
+    }
+
     for (int i = 0; i < count; i++) {
-        entries[i] = strdup(objects[i].c_str());
+        if ((entries[i] = strdup(objects[i].c_str())) == NULL) {
+            s3->log->LogVerboseMessage(
+              -3, "S3ObjectListAdd: unable to allocate memory for object.");
```

Comment:
Could this message be clearer? Maybe "Unable to allocate memory for the object string"?